libpod/container_inspect.go

```diff
@@ -0,0 +1,161 @@
+package libpod
+
+import (
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/projectatomic/libpod/libpod/driver"
+	"github.com/sirupsen/logrus"
+)
+
+func getContainerData(ctr *Container, size bool, driverData *driver.Data) (*ContainerData, error) {
+	config := ctr.config
+	runtimeInfo := ctr.state
+	spec := ctr.config.Spec
+
+	cpus, mems, period, quota, realtimePeriod, realtimeRuntime, shares := getCPUInfo(spec)
+	blkioWeight, blkioWeightDevice, blkioReadBps, blkioWriteBps, blkioReadIOPS, blkioeWriteIOPS := getBLKIOInfo(spec)
+	memKernel, memReservation, memSwap, memSwappiness, memDisableOOMKiller := getMemoryInfo(spec)
+	pidsLimit := getPidsInfo(spec)
+
+	args := config.Spec.Process.Args
+	path := args[0]
+	if len(args) > 0 {
+		args = args[1:]
+	}
+
+	data := &ContainerData{
+		Id:      config.ID,
+		Created: config.CreatedTime,
+		Path:    path,
+		Args:    args,
+		State: &CtrState{
+			OciVersion: spec.Version,
+			Status:     runtimeInfo.State.String(),
+			Running:    runtimeInfo.State == ContainerStateRunning,
+			Paused:     runtimeInfo.State == ContainerStatePaused,
+			OOMKilled:  runtimeInfo.OOMKilled,
+			Dead:       runtimeInfo.State.String() == "bad state",
+			Pid:        runtimeInfo.PID,
+			ExitCode:   runtimeInfo.ExitCode,
+			Error:      "", // can't get yet
+			StartedAt:  runtimeInfo.StartedTime,
+			FinishedAt: runtimeInfo.FinishedTime,
+		},
+		ImageID:         config.RootfsImageID,
+		ImageName:       config.RootfsImageName,
+		ResolvConfPath:  runtimeInfo.ConfigPath,                               // not sure
```

Comment:
Can't get this yet, coming in networking patch - leave as "" for now

Reply:
for the "" that are pending, can you add a TODO comment or perhaps PENDING.

Reply:
fixed